homeassistant/components/soma/config_flow.py

```diff
@@ -40,14 +40,22 @@ def __init__(self):
         """Finish config flow."""
         api = SomaApi(user_input["host"], user_input["port"])
         try:
-            await self.hass.async_add_executor_job(api.list_devices)
+            result = await self.hass.async_add_executor_job(api.list_devices)
             _LOGGER.info("Successfully set up Soma Connect")
-            return self.async_create_entry(
-                title="Soma Connect",
-                data={"host": user_input["host"], "port": user_input["port"]},
+            if result["result"] == "success":
+                return self.async_create_entry(
+                    title="Soma Connect",
+                    data={"host": user_input["host"], "port": user_input["port"]},
+                )
+            _LOGGER.error(
+                "Connection to SOMA Connect failed (result:%s)" % (result["result"])
             )
```

Comment:
The [pylint error](https://dev.azure.com/home-assistant/Home%20Assistant/_build/results?buildId=19702&view=logs&j=8b66295a-05e1-52cf-5640-7a4f4e1975e2&t=6c4465fe-285a-5650-787c-2e3dbe1da192&l=13) seems to be about these lines. I would recommend this small fix:
```python
_LOGGER.error(
    "Connection to SOMA Connect failed (result: %s)",  result["result"]
)
```